src/topology/connected.lean

```diff
@@ -198,25 +198,48 @@ def connected_component_in (F : set α) (x : F) : set α := coe '' (connected_co
 theorem mem_connected_component {x : α} : x ∈ connected_component x :=
 mem_sUnion_of_mem (mem_singleton x) ⟨is_connected_singleton.is_preconnected, mem_singleton x⟩
 
+theorem is_preconnected_connected_component {x : α} : is_preconnected (connected_component x) :=
+is_preconnected_sUnion x _ (λ _, and.right) (λ _, and.left)
+
 theorem is_connected_connected_component {x : α} : is_connected (connected_component x) :=
-⟨⟨x, mem_connected_component⟩, is_preconnected_sUnion x _ (λ _, and.right) (λ _, and.left)⟩
+⟨⟨x, mem_connected_component⟩, is_preconnected_connected_component⟩
 
-theorem subset_connected_component {x : α} {s : set α} (H1 : is_preconnected s) (H2 : x ∈ s) :
-  s ⊆ connected_component x :=
+theorem is_preconnected.subset_connected_component {x : α} {s : set α}
+  (H1 : is_preconnected s) (H2 : x ∈ s) : s ⊆ connected_component x :=
 λ z hz, mem_sUnion_of_mem hz ⟨H1, H2⟩
 
+theorem is_connected.subset_connected_component {x : α} {s : set α}
+  (H1 : is_connected s) (H2 : x ∈ s) : s ⊆ connected_component x :=
+H1.2.subset_connected_component H2
+
+theorem connected_component_eq {x y : α} (h : y ∈ connected_component x) :
+  connected_component x = connected_component y :=
+eq_of_subset_of_subset
+  (is_connected_connected_component.subset_connected_component h)
+  (is_connected_connected_component.subset_connected_component
+    (set.mem_of_mem_of_subset mem_connected_component
+      (is_connected_connected_component.subset_connected_component h)))
+
+lemma connected_component_disjoint {x y : α} (h : connected_component x ≠ connected_component y) :
+  disjoint (connected_component x) (connected_component y) :=
+set.disjoint_left.2 (λ a h1 h2, h
+  ((connected_component_eq h1).trans (connected_component_eq h2).symm))
+
 theorem is_closed_connected_component {x : α} :
   is_closed (connected_component x) :=
 closure_eq_iff_is_closed.1 $ subset.antisymm
-  (subset_connected_component
-    is_connected_connected_component.closure.is_preconnected
+  (is_connected_connected_component.closure.subset_connected_component
     (subset_closure mem_connected_component))
   subset_closure
 
+lemma continuous.image_connected_component_subset {β : Type*} [topological_space β] {f : α → β}
```

Comment:
I put this in the `continuous` namespace for dot notation, but this could also be done for a bunch of the new lemmas below as well. Would that be helpful or is it going overboard?

Reply:
I think it would be helpful, but I dont exactly know which. Many of the lemmas that need a continuous map also require the codomain to be totally disconnected, should those still go in the continuous namespace?

Reply:
`totally_disconnected_space α` seems to be a type class argument in most of those, so I think `continuous` might still be appropriate.

I'll push a commit with some more changes shortly.